biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceBuilder.java

```diff
@@ -1010,10 +1045,9 @@ public static SupportingResource parse(File file, URI uri) {
 			uri = file.toURI();
 
 		try (Jar jar = new Jar(file)) {
-			ResourceBuilder rb = new ResourceBuilder();
-			boolean hasIdentity = rb.addJar(jar);
+			ResourceBuilder rb = new ResourceBuilder(processor);
 
-			String mime = hasIdentity ? MIME_TYPE_BUNDLE : MIME_TYPE_JAR;
+			String mime = jar.getBsn() != null ? MIME_TYPE_BUNDLE : MIME_TYPE_JAR;
```

Comment:
The problem that we have is that generating the content capability requires us to know whether the jar is a bundle. If we generate that capability *after* adding the jar then we can't copy it into the supporting resources which need access to the URI and SHA.

Reply:
The idea was that you add the content capability when you know you have a URL & a file. Putting a requirement to pass a content capability present in `addJar()` is an incompatible change.  I've been struggling with this when I cleaned up the code.

Since your solution requires the URL, I think you need to encode the MRJ in this method. I would just call `addJar(jar)` for the primary resource as previous. (The MRJ code there should be gated by the the settings)

And then if you have identity & it is an MRJ & the settings is URL mangling then you create the supporting resources. Would probably simplify your code and would not require passing the URL & SHA under the surface.

The synthetic solution does not require the URL so it can be fully handled on the `addJar(jar)` method. This can be useful for testing.